api.ts

```diff
@@ -267,7 +268,9 @@ export function wrapMetadataFunction(
 /** Options you can specify when defining a parameter using {@link makeParameter}. */
 export type ParameterOptions<T extends ParameterType> = Omit<ParamDef<ParameterTypeMap[T]>, 'type' | 'autocomplete'> & {
   type: T;
-  autocomplete?: MetadataFormulaDef | Array<string | number | SimpleAutocompleteOption>;
+  autocomplete?: T extends ParameterType.Number | ParameterType.String
```

Comment:
Does this mean we won't allow you to define an autocomplete for a parameter that's not a string or a number? I guess that's consistent with what we technically support right now, just wonder if it will break anybody who did it for a boolean or date field and it kinda works for whatever reasons due to quirks of CFL. But I guess that's ok if we don't intend for that to work.

Reply:
yeah, I thought about it. It's unclear what the expected autocomplete return type would be if the parameter type isn't string or number. So I just restrict this to number / string. 

It won't break existing pack bundles but would give TS errors if non string/number parameter has an autocomplete. I'll verify the packs repo first before I merge this PR.

Reply:
tested in packs repo that make validate works.